Plugin.cs

```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
             if (!this.configuration.alwaysOn && (!this.configuration.smart || !Retainer())) return;
             var listing = MarketBoardCurrentOfferings.Read(dataPtr);
             var i = 0;
-            if (this.configuration.hq && this.items.Single(j => j.RowId == listing.ItemListings[0].CatalogId).CanBeHq)
-            {
+
+            // Load addon ptr from memory
+            var toolTipPtr = pi.Framework.Gui.GetUiObjectByName("ItemDetail", 1) + 0x258;
+            var toolTipItemName = GetSeStringText(GetSeString(toolTipPtr));
+            bool isCurrentItemHQ = false;
+
+            // Checks for HQ symbol in item name
+            if (toolTipItemName.Substring(toolTipItemName.Length - 1) == "")
+                isCurrentItemHQ = true;
+            else
+                isCurrentItemHQ = false;
+
+            if (isCurrentItemHQ) {
                 while (i < listing.ItemListings.Count && !listing.ItemListings[i].IsHq) i++;
                 if (i == listing.ItemListings.Count) return;
+            } else {
+                while (i < listing.ItemListings.Count && listing.ItemListings[i].IsHq) i++;
+                if (i == listing.ItemListings.Count) return;
             }
+
             var price = listing.ItemListings[i].PricePerUnit - this.configuration.delta;
             Clipboard.SetText(price.ToString());
-            if (this.configuration.verbose) this.pi.Framework.Gui.Chat.Print($"{price} copied to clipboard.");
+            if (this.configuration.verbose) {
+                if (isCurrentItemHQ)
+                    this.pi.Framework.Gui.Chat.Print($"[HQ] {price} copied to clipboard.");
+                else
+                    this.pi.Framework.Gui.Chat.Print($"{price} copied to clipboard.");
+            }
             this.newRequest = false;
         }
+
+        /// <summary>
+        /// Text from addon to SeString
+        /// </summary>
+        /// <param name="textPtr">addon ptr</param>
+        /// <returns></returns>
+        private SeString GetSeString(IntPtr textPtr) {
+            var size = 0;
+            while (Marshal.ReadByte(textPtr, size) != 0)
+                size++;
+
+            var bytes = new byte[size];
+            Marshal.Copy(textPtr, bytes, 0, size);
+
+            return GetSeString(bytes);
+        }
+
+        /// <summary>
+        /// SeString to string
+        /// </summary>
+        /// <param name="sestring"></param>
+        /// <returns></returns>
+        private string GetSeStringText(SeString sestring) {
+            var pieces = sestring.Payloads.OfType<TextPayload>().Select(t => t.Text);
+            var text = string.Join("", pieces).Replace('\n', ' ').Trim();
+            return text;
+        }
+
+        /// <summary>
+        /// Parse bytes to SeString
+        /// </summary>
+        /// <param name="bytes"></param>
+        /// <returns></returns>
+        private SeString GetSeString(byte[] bytes) {
+            return pi.SeStringManager.Parse(bytes);
+        }
```

Comment:
Is this copied from somewhere? Not a problem if it is, just curious what the original source is.

Reply:
Yes
https://github.com/daemitus/YesAlready/blob/master/YesAlready/YesAlreadyPlugin.cs#L119